src/main/haskell/kore/src/Kore/Builtin/KEqual.hs

```diff
@@ -113,3 +154,44 @@ evalKEq true false tools substitutionSimplifier _ pat =
       where
         ep1 = ExpandedPattern.fromPurePattern t1
         ep2 = ExpandedPattern.fromPurePattern t2
+
+evalKIte
+    ::  forall variable
+    .   ( FreshVariable variable
+        , Hashable variable
+        , OrdMetaOrObject variable
+        , SortedVariable variable
+        , ShowMetaOrObject variable
+        )
+    => MetadataTools.MetadataTools Object StepperAttributes
+    -> PredicateSubstitutionSimplifier Object Simplifier
+    -> PureMLPatternSimplifier Object variable
+    -> Application Object (PureMLPattern Object variable)
+    -> Simplifier
+        ( AttemptedFunction Object variable
+        , SimplificationProof Object
+        )
+evalKIte _ _ _ pat =
+    case pat of
+        Application
+            {  applicationChildren = [expr, t1, t2] } ->
```

Comment:
This must throw `Builtin.wrongArity` if called with the wrong arity. Although the definition and pattern are verified, it is possible that the pattern was rewritten in the meantime. If the arity is wrong, we must not return `notApplicableFunctionEvaluator`.